src/modules/deal-guarantee/dto/get-deal-guarantee-params.dto.ts

```diff
@@ -0,0 +1,10 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { EXAMPLES } from '@ukef/constants';
+import { UkefId } from '@ukef/helpers';
+import { Matches } from 'class-validator';
+
+export class GetDealsGuaranteesParamsDto {
+  @ApiProperty({ description: 'UKEF id for deal', example: EXAMPLES.DEAL_ID })
+  @Matches(/00\d{8}/)
```

Comment:
I've also commented [here](https://github.com/UK-Export-Finance/tfs-api/pull/73#discussion_r1165603593) but I think we should drop the validation on the URL field because we should be returning a 404 anyway since the party won't exist in ACBS - what do you think?

Reply:
I would prefer to have Deal/Facility/UKEF id validations. It was missing in Mulesoft implementation and it caused some issues. It is easy to implement.
Deal Guarantee endpoint is not so important, but GET Deal and GET Facility are used to check if id is free and can be used, 404 is expected positive response, 400 would mean real issue. We had a bug where id 'pending' was used instead of 10 digit string.

Reply:
Pear review - lets test it user input for ukef id's, and return 400.